runtime/pkg/rduckdb/db.go

```diff
@@ -578,7 +654,23 @@ func (d *db) RenameTable(ctx context.Context, oldName, newName string) error {
 	if strings.EqualFold(oldName, newName) {
 		return fmt.Errorf("rename: Table with name %q already exists", newName)
 	}
-	err := d.writeSem.Acquire(ctx, 1)
+
+	// track write for both old and new table
+	err := d.trackWrite(oldName)
+	if err != nil {
+		return err
+	}
+	err = d.trackWrite(newName)
+	if err != nil {
+		d.untrackWrite(oldName)
+		return err
+	}
+	defer func() {
+		d.untrackWrite(oldName)
+		d.untrackWrite(newName)
+	}()
```

Comment:
Would this be simpler?

```suggestion
	err := d.trackWrite(oldName)
	if err != nil {
		return err
	}
	defer d.untrackWrite(oldName)
	err = d.trackWrite(newName)
	if err != nil {
		return err
	}
	defer d.untrackWrite(newName)
```